File: programming-languages/exercises/cpp/simple_calculator.cpp
#include <iostream>
#include <string>
using namespace std;
int main(){
    int num1, num2;
    char op;
    while(1){
        cout << "Enter operator (q to quit): " << endl;
        cin >> op;
        if(op == 'q')break;
        cout << "Enter num1 num2: \n";
        cin >> num1 >> num2;
        switch(op){
            case '+':
                printf("%d\n", (num1+num2));
                break;
            case '-':
                printf("%d\n", (num1-num2));
                break;
            case '*':
                printf("%d\n", (num1*num2));
                break;
            case '/':
                if (num2==0){
                    printf("cannot divide by zero \n");
                }else{
                    printf("%d\n", (num1/num2));
                }
                break;
            case '%':
                printf("%d\n", (num1%num2));
                break;
        }
    }
}

File: programming-languages/exercises/python/simple_calculator.py
while true:
    inputs=input("Enter num1 num2 (or q to quit): ").split()
    if "q" in inputs:
        break
    op=input("Enter Operator: ")
    num_list=list(map(int, inputs))
    match op:
        case "+":
            print(num_list[0] + num_list[1])
        case "-":
            print(num_list[0] - num_list[1])
        case "*":
            print(num_list[0] * num_list[1])
        case "/":
            if num_list[1] == 0:
                print("Cannot divide by zero")
            else:
                print(num_list[0] / num_list[1])
        case _:
            print("Invalid operator")